packages/@expo/cli/src/start/server/metro/createExpoMetroResolver.ts

```diff
@@ -180,15 +182,24 @@ export function createFastResolver({
           // Disable `browser` field for server environments.
           isServer
             ? undefined
-            : // Enable `browser` field support
+            : // Enable package.json `browser` and `react-native` fields support
               (pkg: any, _resolvedPath: string, relativePathIn: string): string => {
                 let relativePath = relativePathIn;
                 if (relativePath[0] !== '.') {
                   relativePath = `./${relativePath}`;
                 }
 
-                const replacements = pkg.browser;
```

Comment:
`browser` field and `react-native` fields aren't similar afaict. [spec](nctzombie/package-browser-field-spec). The `mainFields` don't appear to have the same capabilities as the browser field.

Reply:
@EvanBacon Link doesn't seem to work (404)

I tried to copy behaviour of default Metro resolver which I believe shouldn't be modified it this should be drop-in replacement.